server/impl/src/main/java/com/walmartlabs/concord/server/process/queue/dispatcher/Dispatcher.java

```diff
@@ -175,7 +175,16 @@ private List<Match> match(DSLContext tx, List<Request> requests) {
             // filter out the candidates that shouldn't be dispatched at the moment (e.g. due to concurrency limits)
             for (ProcessQueueEntry e : candidates) {
                 // find request/agent who can handle process
-                Request req = findRequest(e, inbox);
+                Request req = findRequest(e, inbox, (queueEntry, ex) -> {
```

Comment:
Make the error handler a class member instead? I.e. init it once in the constructor?